plugins/flatpak/gs-flatpak.c

```diff
@@ -1301,6 +1320,39 @@ gs_flatpak_app_install_source (GsFlatpak *self, GsApp *app,
 	return TRUE;
 }
 
+static GsApp *
+get_main_app_of_related (GsFlatpak *self,
+			 GsApp *related_app,
+			 GCancellable *cancellable,
+			 GError **error)
+{
+	g_autofree gchar *main_app_name = g_strdup (gs_flatpak_app_get_ref_name (related_app));
+	g_autoptr(FlatpakInstalledRef) ref = NULL;
+	const gchar *app_name;
+	g_auto(GStrv) app_tokens = NULL;
+
+        app_name = gs_flatpak_app_get_main_app (related_app);
+	if (app_name == NULL)
+		return NULL;
+
+	app_tokens = g_strsplit (app_name, "/", -1);
+	if (g_strv_length (app_tokens) != 4)
+		return NULL;
+
+	ref = flatpak_installation_get_installed_ref (self->installation,
+						      FLATPAK_REF_KIND_APP,
+						      app_tokens[1],
+						      app_tokens[2],
+						      app_tokens[3],
+						      cancellable,
+						      error);
+
+	if (ref == NULL)
```

Comment:
Same comment, need to set GError

Reply:
The `GError` is actually already set in this case.

Reply:
Oh, right! Sorry for the noise